pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +94,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthProvider specifies custom authentication for the kubernetes cluster.
+	AuthProvider *AuthProviderConfig `json:"auth-provider,omitempty"`
```

Comment:
> Good point. I can see this being the case for us over the next few months. Were you thinking that this could be a slice instead?

Yes, with the godoc describing the wrapping order and warning that we provide no protection against one wrapper undoing the work of another wrapper.